# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/SSLServerCertVerification.cpp
# Commit: dc509eab3ed7
# Full Hash: dc509eab3ed7a513258723eb46b25f33a610e720
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2025-07-11 16:36:21
# Regressor Bug: 1953221
# File Overlap Count: 3
# Description:
#   Bug 1953221 - don't accept SCTs with a timestamp after the distrustAfter date for the root r=jschanck
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D256720
# ==============================================================================

diff -r 5b86ae2c81f9 -r dc509eab3ed7 security/manager/ssl/SSLServerCertVerification.cpp
--- a/security/manager/ssl/SSLServerCertVerification.cpp	Thu Jul 10 18:37:49 2025 +0000
+++ b/security/manager/ssl/SSLServerCertVerification.cpp	Thu Jul 10 19:04:10 2025 +0000
@@ -481,6 +481,9 @@
   for (size_t i = 0; i < info.verifyResult.sctsWithInvalidTimestamps; ++i) {
     glean::ssl::scts_verification_status.AccumulateSingleSample(4);
   }
+  for (size_t i = 0; i < info.verifyResult.sctsWithDistrustedTimestamps; ++i) {
+    glean::ssl::scts_verification_status.AccumulateSingleSample(6);
+  }
 
   // See scts_origin in metrics.yaml.
   for (size_t i = 0; i < info.verifyResult.embeddedSCTs; ++i) {